### PR TITLE
sensors: Remove unnecessary Kconfig setting of sensors

### PR DESCRIPTION
**Pull request by galak**
Sensor Kconfig sybmols should be enabled if CONFIG_SENSOR=y
and the devicetree node for the sensor is enabled.  We can
remove explicitly enabling specific sensor drivers in .conf
files.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Comment by galak** [comment]
@bbilas here's the PR to remove the Kconfig symbols for sensors :)

File: samples/boards/96b_argonkey/sensors/prj.conf [hunk 1]
```diff
@@ -6,9 +6,5 @@ CONFIG_LED=y
 CONFIG_GPIO=y
 CONFIG_SENSOR=y
 CONFIG_SENSOR_LOG_LEVEL_DBG=y
-CONFIG_HTS221=y
-CONFIG_LPS22HB=y
-CONFIG_VL53L0X=y
-CONFIG_LSM6DSL=y
 CONFIG_LP3943=y
```
Comment (bbilas) [inline samples/boards/96b_argonkey/sensors/prj.conf:9]: What about that one?

File: samples/sensor/adxl372/prj.conf [hunk 2]
```diff
@@ -2,7 +2,6 @@ CONFIG_STDOUT_CONSOLE=y
 CONFIG_LOG=y
 CONFIG_SPI=y
 CONFIG_SENSOR=y
-CONFIG_ADXL372=y
 CONFIG_ADXL372_SPI=y
```
Comment (bbilas) [inline samples/sensor/adxl372/prj.conf:5]: btw. we should also get rid of those kconfig symbols and use some macros to determine the bus but that would be a case for another PR.

File: samples/boards/96b_argonkey/sensors/prj.conf (hunk 1 above)
Comment (galak) [reply samples/boards/96b_argonkey/sensors/prj.conf:9]: Its not a sensor, its a LED and hasn't been converted over to the new Kconfig logic/style.

File: samples/sensor/adxl372/prj.conf (hunk 2 above)
Comment (galak) [reply samples/sensor/adxl372/prj.conf:5]: We can rework the driver to not need the Kconfig symbols and do what other drivers do:
`DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)` and `DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)`

File: samples/boards/96b_argonkey/sensors/prj.conf (hunk 1 above)
Comment (bbilas) [reply samples/boards/96b_argonkey/sensors/prj.conf:9]: Ah right, I've looked too briefly :)

**Review by bbilas — DISMISSED** [review]

**Review by MaureenHelm — COMMENTED** [review]
Why is this DNM?

**Comment by galak** [comment]
> Why is this DNM?

Marked it DNM as no twister CI was run due to it impacting too many tests.  @nashif posted a fix for CI here - https://github.com/zephyrproject-rtos/zephyr/pull/48168.  So once that gets merged, will rebase so we get some actual build tests going.

**Comment by MaureenHelm** [comment]
> > Why is this DNM?
> 
> Marked it DNM as no twister CI was run due to it impacting too many tests. @nashif posted a fix for CI here - #48168. So once that gets merged, will rebase so we get some actual build tests going.

Good catch

**Review by MaureenHelm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by bbilas — APPROVED** [review]
